src/main/java/org/spongepowered/api/plugin/PluginContainer.java

```diff
@@ -52,10 +57,19 @@
     String getVersion();
 
     /**
-     * Returns the created instance of {@link Plugin}.
+     * Returns the assigned logger to this {@link Plugin}.
+     *
+     * @return The assigned logger
+     */
+    default Logger getLogger() {
+        return LoggerFactory.getLogger(getId());
+    }
+
+    /**
+     * Returns the created instance of {@link Plugin} if it is available.
      *
-     * @return The instance
+     * @return The instance if available
      */
-    Object getInstance();
+    Optional<Object> getInstance();
```

Comment:
Forge doesn't guarantee that there is a instance attached for all mod containers. It provides several "dummy" mod containers which don't ever get an attached mod instance, e.g. the MCP container, the Minecraft container and even the FML and Minecraft Forge container.

Additionally the mod containers are implemented by core mods manually, so they could theoretically return anything there. I just checked a few core mods and only some of them actually provide the instance. Currently we even check in the `/sponge plugins` command [if the provided instance is null](https://github.com/SpongePowered/SpongeCommon/blob/master/src/main/java/org/spongepowered/common/command/CommandSponge.java#L382-L383), which is - in my opinion - definitely not a good way considering as per specification of our API it should never return null.